art_internal_impl.hpp

```diff
@@ -1696,6 +2136,98 @@ class basic_inode_48 : public basic_inode_48_parent<ArtPolicy> {
+  // lexicographic ordering that is mapped to some child.  We scan
+  // child_indexes[256], which contains an index into each of the 48
+  // possible children iff a given key is mapped to a child, looking
+  // for the first mapped entry. That will be the smallest key mapped
+  // by the N48 node.
+  [[nodiscard]] constexpr typename basic_inode_48::iter_result begin() noexcept {
+    for ( std::uint64_t i = 0; i < 256; i++ ) { // TODO Is there a declared constant for the child_indexes[] capacity?
+      if ( child_indexes[ i ] != empty_child ) {
+        const std::byte key = static_cast<std::byte>( i );
+        const std::uint8_t child_index = static_cast<std::uint8_t>( i ); // downcast is safe since in [0:255]
+        return { node_ptr{ this, node_type::I48 }, key, child_index };
+      }
+    }
+    UNODB_DETAIL_CANNOT_HAPPEN(); // because we always have at least 17 keys.
+  }
+  
+  // N48: Return the child pointer for the last key in the
+  // lexicographic ordering that is mapped to some child.  We scan
+  // child_indexes[256] in reverse, which contains an index into each
+  // of the 48 possible children iff a given key is mapped to a child,
+  // looking for the last mapped entry. That will be the greatest key
+  // mapped by the N48 node.
+  [[nodiscard]] constexpr typename basic_inode_48::iter_result last() noexcept {
+    for ( std::int64_t i = 255; i >= 0; i-- ) { // TODO Is there a declared constant for the child_indexes[] capacity?
+      if ( child_indexes[ static_cast<std::uint8_t>( i ) ] != empty_child ) { // downcase is safe since in [0:255]
+        const std::byte key = static_cast<std::byte>( i );
+        const std::uint8_t child_index = static_cast<std::uint8_t>( i ); // downcast is safe since in [0:255]
+        return { node_ptr{ this, node_type::I48 }, key, child_index };
+      }
+    }
+    UNODB_DETAIL_CANNOT_HAPPEN(); // because we always have at least 17 keys.
+  }
+  
+  [[nodiscard]] constexpr typename basic_inode_48::iter_result_opt next(const std::uint8_t child_index) noexcept {
+    // loop over the remaining byte values in lexical order.
+    for ( std::uint64_t i = static_cast<std::uint64_t>(child_index) + 1; i < 256; i++ ) {
+      if ( child_indexes[ i ] != empty_child ) {
+        const std::byte key = static_cast<std::byte>( i );
+        const std::uint8_t next_index = static_cast<std::uint8_t>( i ); // downcast is safe since in [0:255]
+        return { { node_ptr{ this, node_type::I48 }, key, next_index } };
+      }
+    }
+    return parent_class::end_result;
+  }
+  
+  [[nodiscard]] constexpr typename basic_inode_48::iter_result_opt prior(const std::uint8_t child_index) noexcept {
+    // loop over the prior byte values in lexical order.
+    for ( std::int64_t i = static_cast<std::int64_t>(child_index) - 1; i >= 0; i-- ) {
+      if ( child_indexes[ static_cast<std::uint8_t>(i) ] != empty_child ) { // downcast ok since in [0:255]
+        const std::byte key = static_cast<std::byte>( i );
```

Comment:
Here and everywhere these two lines repeat, `auto`

Reply:
done